scrapi/harvesters/vivo.py

```diff
@@ -0,0 +1,216 @@
+                                    }}
+                            OPTIONAL {{
+                                        ?URI bibo:volume ?volume .
+                                    }}
+                            OPTIONAL {{
+                                        ?URI bibo:number ?number .
+                                    }}
+                            OPTIONAL {{
+                                        ?URI bibo:pageStart ?startPage .
+                                    }}
+                            OPTIONAL {{
+                                        ?URI bibo:pageEnd ?endPage .
+                                    }}
+                            OPTIONAL {{
+                                        ?URI bibo:pmid ?PMID .
+                                   }} 
+                            OPTIONAL {{
+                                        ?autorship a vivo:Authorship .
+                                        ?autorship vivo:relates ?URI .
+                                        ?authorURI vivo:relatedBy ?autorship .
+                                        ?authorURI a foaf:Person .
+                                        ?authorURI rdfs:label ?authorName .
+                                        BIND ( CONCAT(str(?authorURI), ';' , ?authorName) AS ?author) .
+                                    }}
+                            FILTER (strdt(?date, xsd:date) >= "{}"^^xsd:date && strdt(?date, xsd:date) <= "{}"^^xsd:date) 
+                        }} GROUP BY ?URI ?doi ?abstract ?title ?journalTitle ?date ?publisher ?volume ?number ?startPage ?endPage ?PMID ?ISSN ?ISBN LIMIT {} OFFSET {}
+                    """
+        query_str = query_str.format(start_date.isoformat(), end_date.isoformat(), limit, offset)
+        self.sparql_wrapper.setQuery(query_str)
+        results = self.sparql_wrapper.query()
+        results = results.convert()
+        return results['results']['bindings']
+
+    @property
+    def schema(self):
+        return {
+            'title': ('/title', lambda x: x if x else ''),
+            'description': ('/subtitle', lambda x: x[0] if (isinstance(x, list) and x) else x or ''),
+            'providerUpdatedDateTime': ('/date', lambda x: datetime.strptime(x, "%Y-%m-%d").strftime("%Y-%m-%dT%H:%M:%S%Z")+'+00:00'),
+            'uris': {
+                'canonicalUri': '/URI'
+            },
+            'contributors': ('/authors', lambda x: process_contributors(x) if x else []),
+            'subjects': ('/subjects', lambda x: x.split('|') if x else []),
+            'otherProperties': build_properties(
+                ('journalTitle', '/journalTitle'),
+                ('abstract', ('/abstract', lambda x: x if x else '')),
+                ('issue', '/issue'),
+                ('publisher', '/publisher'),
+                ('types', ('/types', lambda x: x.split('|') if x else [])),
+                ('ISSN', ('/ISSN', lambda x: x if x else '')),
```

Comment:
If this can be resolved to a URI, it can fit in one of the URI fields (`uris.objectUris`, `uris.providerUris`, `uris.descriptorUris`). These fields all have descriptions in the schema (let me know if you need help interpreting it! The structure is a little strange.)